src/nb_clean/__init__.py

```diff
@@ -227,6 +227,10 @@ def clean_notebook(
             cell["execution_count"] = None
             if not preserve_cell_outputs:
                 cell["outputs"] = []
+            else:
+                for dict_ in cell["outputs"]:
+                    if "execution_count" in dict_.keys():
```

Comment:
There's no need for `.keys()` here: `if key in dict_:` performs a constant time lookup in the dictionary's keys.

Reply:
I forget that iterate over dict will iterate over it keys. Thank you.